### PR TITLE
Affinities

### PR DESCRIPTION
**Pull request by Oozyx**
Added affinities and price feeds.

### CODE REVIEW

File: contracts/CryptoChampions.sol [hunk 1]
```diff
@@ -109,8 +113,8 @@ contract CryptoChampions is ICryptoChampions, AccessControl, ERC1155 {
         currentPhase = Phase.ONE;
     }
 
-    modifier isValidElderSpiritId(uint elderId) {
-        require(elderId > IN_GAME_CURRENCY_ID && elderId <= MAX_NUMBER_OF_ELDERS);  // dev: Given id is not valid.
```
Comment (cds95) [inline contracts/CryptoChampions.sol:113]: I've added a `modifier` for this in the branch I merged into main.  Should that be used here?

File: contracts/CryptoChampions.sol [hunk 2]
```diff
@@ -427,10 +453,28 @@ contract CryptoChampions is ICryptoChampions, AccessControl, ERC1155 {
...
+        returns (
+            bool,
+            uint256,
+            uint256,
+            string memory,
+            int256
+        )
+    {
         ElderSpirit memory elderSpirit = _elderSpirits[elderId];
-        return (elderSpirit.valid, elderSpirit.raceId, elderSpirit.classId, elderSpirit.affinity);
```
Comment (cds95) [inline contracts/CryptoChampions.sol:434]: Just curious, does Solidity have something like Prettier where files get formatted on save?

File: contracts/CryptoChampions.sol (hunk 1 above)
Comment (Oozyx) [reply contracts/CryptoChampions.sol:113]: Yea I didn't delete the modifier, only the parameter changed from uint to uint256.

File: contracts/CryptoChampions.sol (hunk 2 above)
Comment (Oozyx) [reply contracts/CryptoChampions.sol:434]: Yes it does I have it configured on my system. I can share my vs code settings later if you want.

Comment (cds95) [reply contracts/CryptoChampions.sol:434]: @Oozyx yes please.

**Pull request merged**